subprojects/code-quality/src/main/groovy/org/gradle/api/plugins/quality/CheckstyleExtension.java

```diff
@@ -113,7 +116,18 @@ public File getConfigDir() {
      */
     @Incubating
     public void setConfigDir(File configDir) {
-        this.configDir = configDir;
+        this.configDir.set(configDir);
+    }
+
+    /**
+     * Set the provider for calculating the configuration directory.
+     *
+     * @param configDir Configuration directory provider
+     * @since 4.7
+     */
+    @Incubating
+    public void setConfigDir(Provider<File> configDir) {
```

Comment:
❌ Let's use the recommended idiom for providers: A getter returning the `DirectoryProperty`, so users can call get/set on it as needed. At some point we can then deprecate the old getters/setters.

We'll need a different name of course, could be `getConfigurationDirectory`

Reply:
See comment above.